rfc/text/advanced/ap_10_rpc_progressive_calls.md

```diff
@@ -1,5 +1,249 @@
 ## Progressive Calls {#rpc-progressive-calls}
 
-A procedure implemented by a *Callee* and registered at a *Dealer* may receive a progressive call. Progressive results can e.g. be used to start processing initial data where a larger data set may not yet have been generated or received by the *Caller*.
+A *Caller* may issue a progressive call. This can be useful in few cases:
 
-See this GitHub issue for more discussion: https://github.com/wamp-proto/wamp-proto/issues/167
+* Payload is too big to send it whole in one request. E.g. uploading a file.
+* Payload is long term-based, and it is required to start processing it earlier.
+* Partial results should be made available to the caller as quickly as possible. This can be accomplished
+  in conjunction with progressive call results feature.
```

Comment:
with the Progressive Call Results feature.

Reply:
thnx!